src/modules/rover_pos_control/RoverPositionControl.cpp

```diff
@@ -379,32 +428,76 @@ RoverPositionControl::control_attitude(const vehicle_attitude_s &att, const vehi
 {
 	// quaternion attitude control law, qe is rotation from q to qd
 	const Quatf qe = Quatf(att.q).inversed() * Quatf(att_sp.q_d);
+
+	// Attitude error in Euler angles [rad]
 	const Eulerf euler_sp = qe;
+	// PX4_INFO("Yaw error: %f [rad]", double(euler_sp(2)));
 
-	float control_effort = euler_sp(2) / _param_max_turn_angle.get();
-	control_effort = math::constrain(control_effort, -1.0f, 1.0f);
+	_rates_sp.roll = 0.0;
+	_rates_sp.pitch = 0.0;
 
-	_act_controls.control[actuator_controls_s::INDEX_YAW] = control_effort;
+	// Apply Attitude P-gain for Yaw control. Euler angle in [rad], rate setpoint in [rad/s]
+	_rates_sp.yaw = euler_sp(2) * _param_att_p.get();
 
-	const float control_throttle = att_sp.thrust_body[0];
+	// PX4_INFO("Yaw rate setpoint in attitude control: %f [rad/s]", (double)_rates_sp.yaw);
+
+	_rates_sp.thrust_body[0] = math::constrain(att_sp.thrust_body[0], 0.0f, 1.0f);
+	_rates_sp.timestamp = hrt_absolute_time();
+	_rates_sp_pub.publish(_rates_sp);
+
+	// Log yaw setpoint for PID tuning debug
+	_local_pos_setpoint.yaw = att_sp.yaw_body;
+}
+
+void
+RoverPositionControl::control_rates(const vehicle_angular_velocity_s &rates, const  vehicle_angular_acceleration_s &acc,
+				    const vehicle_local_position_s &local_pos,
+				    const vehicle_rates_setpoint_s &rates_sp)
+{
+	float dt = (_control_rates_last_called > 0) ? hrt_elapsed_time(&_control_rates_last_called) * 1e-6f : 0.01f;
+	_control_rates_last_called = hrt_absolute_time();
+
+	const matrix::Vector3f vehicle_rates(rates.xyz[0], rates.xyz[1], rates.xyz[2]);
+	const matrix::Vector3f rates_setpoint(rates_sp.roll, rates_sp.pitch, rates_sp.yaw);
+
+	// PX4_INFO("Yaw rate setpoint: %f [rad/s]", (double)_rates_sp.yaw);
+
+	const matrix::Vector3f current_velocity(local_pos.vx, local_pos.vy, local_pos.vz);
+	bool lock_integrator = bool(current_velocity.norm() < _param_rate_i_minspeed.get());
+
+	const matrix::Vector3f angular_acceleration{acc.xyz};
+	const matrix::Vector3f torque = _rate_control.update(vehicle_rates, rates_setpoint, angular_acceleration, dt,
+					lock_integrator);
+
+	// Direct mapping from torque setpoint to steering
+	_steering_input = math::constrain(torque(2), -1.0f, 1.0f);
+
+	// _steering_input = math::constrain(_steering_input + torque(2), -1.0f, 1.0f);
+
+	_act_controls.control[actuator_controls_s::INDEX_YAW] = _steering_input;
```

Comment:
As we have already discussed above, this may work for boats but will break rovers and I am not sure if this is the right way of implementing yaw control for rovers. I dont think this should be slipped in without testing on ackermann steering rovers

Reply:
Oh I thought I added a comment about this, but yeah I added this as a temporary commit, for testing purposes first :thinking:

Reply:
How could this be now reimplemented, as the actuator_control topic has been split?